dstep/driver/Application.d

```diff
@@ -40,57 +40,33 @@ class Application
     public void run ()
     {
         import std.exception : enforce;
+        import std.range;
 
         enforce!DStepException(config.inputFiles.length > 0,
             "dstep: error: must supply at least one input file\n");
 
         enforceInputFilesExist(config);
 
-        // when only one input file is supplied, -o argument is
-        // interpreted as file path, otherwise as base directory path
-        bool singleFileInput = config.inputFiles.length == 1;
+        auto inputFiles = config.inputFiles;
+        auto outputFiles = makeOutputFiles(config);
+        auto translationUnits = makeTranslationUnits(config);
 
-        Task!(
-            startParsingFile,
-            Configuration,
-            string,
-            string)*[] conversionTasks;
+        enforceTranslationUnitsCompiled(translationUnits);
 
-        conversionTasks.length = config.inputFiles.length;
-
-        // parallel generation of D modules for each of input files
-        foreach (size_t index, fileName; config.inputFiles)
+        foreach (tuple; taskPool.parallel(zip(inputFiles, outputFiles, translationUnits), 1))
```

Comment:
* I think this would look nicer with UFCS
* I don't think `taskPool` is required, see [1]

[1] https://dlang.org/phobos/std_parallelism.html#.parallel

Reply:
Fixed.